src/model/modifier.rs

```diff
@@ -0,0 +1,49 @@
+use combine::{optional, token, Parser, Stream};
+use core::fmt;
+use strum_macros::EnumIter;
+use Modifier::*;
+
+#[derive(Clone, Copy, Debug, EnumIter, Eq, PartialEq)]
+pub enum Modifier {
+    Flat,
+    Natural,
+}
+
+impl Modifier {
+    pub fn parser<Input>() -> impl Parser<Input, Output = Self>
+    where
+        Input: Stream<Token = char>,
+    {
+        let flat_parser = token('b').map(|_| Flat);
+
+        optional(flat_parser).map(|modifier| modifier.unwrap_or(Natural))
+    }
+}
+
+impl fmt::Display for Modifier {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let string = match self {
+            Flat => "b",
+            Natural => "",
+        };
+
+        f.write_str(string)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use strum::IntoEnumIterator;
+
+    #[test]
+    fn parser() {
+        let parsed: Vec<_> = Modifier::iter()
+            .map(|m| m.to_string())
```

Comment:
Yep, `ToString` has a default implementation for types which implement `Display`:

```rust
#[stable(feature = "rust1", since = "1.0.0")]
impl<T: fmt::Display + ?Sized> ToString for T {
    // A common guideline is to not inline generic functions. However,
    // removing `#[inline]` from this method causes non-negligible regressions.
    // See <https://github.com/rust-lang/rust/pull/74852>, the last attempt
    // to try to remove it.
    #[inline]
    default fn to_string(&self) -> String {
        use fmt::Write;
        let mut buf = String::new();
        buf.write_fmt(format_args!("{}", self))
            .expect("a Display implementation returned an error unexpectedly");
        buf
    }
}
```